### ISSUE TITLE
Some fields are valid to supply when inserting, but are not allowed to be updated.

### ISSUE THREAD

**Issue by benbenwilde** [clamidity/SalesforceMagic#21]
I ran into this issue when creating an object via `Insert` and then updating it later via `Update`. One of the properties is an id pointing to a parent object, and the property cannot be edited after its created (this is just because of a setting on the object in salesforce). Maybe we could add another attribute like `SalesforceInsertOnly` or something similar to deal with this situation. I don't know if thats the best way to go about it or not. Right now I just set it to null when updating and that works ok for now.

**Comment by clamidity** [comment]
I have been thinking about consolidating the attributes into a single configurable attribute. The individual attributes would still be supported but deprecated.

**Comment by zendever** [comment]
+1

It becomes difficult with the SalesforceIgnore attribute also affecting the reads.

**Comment by benbenwilde** [comment]
Yeah I was setting it to null but that broke cuz then the null handling got fixed (setting the salesforce field to null when you pass null) which is obviously good actually.

So I just use different objects for read and write for now.

**Comment by clamidity** [comment]
Currently there is a SalesforceReadonly attribute that might help, but I am currently working on a single attribute that will have configurable options, also making objects cleaner.

**Comment by derekforeman** [comment]
is SalesforceIgnore working in the current release? I'm using it, but it isn't ignoring..
